misk-zookeeper/src/main/kotlin/misk/zookeeper/CuratorFrameworkProvider.kt

```diff
@@ -47,6 +68,24 @@ internal class CuratorFrameworkProvider @Inject internal constructor(
           }
           ZooKeeper(connectString, sessionTimeout, watcher, canBeReadOnly, clientConfig)
         }
+        .aclProvider(object : ACLProvider {
+          override fun getDefaultAcl(): List<ACL> {
+            // Default ACL allows clients CRUD operations on znodes this service has created.
+            // Service identification comes from the client cert
+            return Collections.singletonList(ACL(DEFAULT_PERMS, AUTH_IDS))
+          }
+
+          override fun getAclForPath(path: String?): List<ACL> {
+            // Shared directories need to be created in such a way that anyone can create them if
+            // they're missing and read or write to them, but anyone should not be able to delete
+            // them.
+            if (path == LEASES_NODE.asZkPath || path == SERVICES_NODE.asZkPath) {
```

Comment:
By defining the `aclProvider` on the `CuratorFrameworkFactory.Builder`, we are sharing this behavior everywhere since the CuratorFramework wraps all interactions with ZooKeeper.

If the path is neither of these then the default ACL is used meaning the node will be created with the default permissions of "the service that created this node can perform CRUD operations".